sound/soc/sof/ipc4-pcm.c

```diff
@@ -79,21 +80,27 @@ sof_ipc4_add_pipeline_to_trigger_list(struct snd_sof_dev *sdev, int state,
 		 * Trigger pipeline if all PCMs containing it are paused or if it is RUNNING
 		 * for the first time
 		 */
-		if (spipe->started_count == spipe->paused_count)
+		if (spipe->started_count == spipe->paused_count) {
+			pipe_priority[trigger_list->count] = pipeline->priority;
 			trigger_list->pipeline_instance_ids[trigger_list->count++] =
 				pipe_widget->instance_id;
+		}
 		break;
 	case SOF_IPC4_PIPE_RESET:
 		/* RESET if the pipeline is neither running nor paused */
-		if (!spipe->started_count && !spipe->paused_count)
+		if (!spipe->started_count && !spipe->paused_count) {
+			pipe_priority[trigger_list->count] = pipeline->priority;
```

Comment:
we are aiming outside of the allocated area?
```c
pipe_priority = kzalloc(pipeline_list->count, GFP_KERNEL);
```
so the last valid member is `pipe_priority[trigger_list->count - 1]`, no?

Some explanation would help to understand why you do this... Set a specific (likely last) priority to `pipeline->priority`

Reply:
@ujfalusi this is OK and not different from the line below

Reply:
Oh, right, **pipeline_list**->count VS **trigger_list**->count